packages/typespec-ts/test/modularUnit/scenarios/models/azureCoreErrorModels.md

```diff
@@ -0,0 +1,515 @@
+  /** The error message. */
+  readonly message?: string;
+  /** The error target. */
+  readonly target?: string;
+  /** The error details. */
+  readonly details?: ErrorDetail[];
+  /** The error additional info. */
+  readonly additionalInfo?: ErrorAdditionalInfo[];
+}
+
+export function errorDetailDeserializer(item: any): ErrorDetail {
+  return {
+    code: item["code"],
+    message: item["message"],
+    target: item["target"],
+    details: !item["details"]
+      ? item["details"]
+      : errorDetailArrayDeserializer(item["details"]),
+    additionalInfo: !item["additionalInfo"]
+      ? item["additionalInfo"]
+      : errorAdditionalInfoArrayDeserializer(item["additionalInfo"]),
+  };
+}
+
+export function errorDetailArrayDeserializer(
+  result: Array<ErrorDetail>,
+): any[] {
+  return result.map((item) => {
+    return errorDetailDeserializer(item);
+  });
+}
+
+export function errorAdditionalInfoArrayDeserializer(
+  result: Array<ErrorAdditionalInfo>,
+): any[] {
+  return result.map((item) => {
+    return errorAdditionalInfoDeserializer(item);
+  });
+}
+
+/** The resource management error additional info. */
+export interface ErrorAdditionalInfo {
+  /** The additional info type. */
+  readonly type?: string;
+  /** The additional info. */
+  readonly info?: any;
+}
+
+export function errorAdditionalInfoDeserializer(
+  item: any,
+): ErrorAdditionalInfo {
+  return {
+    type: item["type"],
+    info: item["info"],
+  };
+}
+
+/** Concrete proxy resource types can be created by aliasing this type using a specific property type. */
+export interface AvsSummary extends ProxyResource {
+  /** The resource-specific properties for this resource. */
+  properties?: AvsSummaryProperties;
+}
+
+export function avsSummarySerializer(item: AvsSummary): any {
+  return {
+    properties: !item["properties"]
+      ? item["properties"]
+      : avsSummaryPropertiesSerializer(item["properties"]),
+  };
+}
+
+export function avsSummaryDeserializer(item: any): AvsSummary {
+  return {
+    id: item["id"],
+    name: item["name"],
+    type: item["type"],
+    systemData: !item["systemData"]
+      ? item["systemData"]
+      : systemDataDeserializer(item["systemData"]),
+    properties: !item["properties"]
+      ? item["properties"]
+      : avsSummaryPropertiesDeserializer(item["properties"]),
+  };
+}
+
+/** model interface AvsSummaryProperties */
+export interface AvsSummaryProperties {
+  error: ErrorDetail_1;
+}
+
+export function avsSummaryPropertiesSerializer(
+  item: AvsSummaryProperties,
+): any {
+  return { error: errorDetailSerializer(item["error"]) };
+}
+
+export function avsSummaryPropertiesDeserializer(
+  item: any,
+): AvsSummaryProperties {
+  return {
+    error: errorDetailDeserializer_1(item["error"]),
+  };
+}
+
+/** model interface ErrorDetail */
+export interface ErrorDetail_1 {
+  code: string;
+  message: string;
+  details?: string;
+}
+
+export function errorDetailSerializer(item: ErrorDetail_1): any {
```

Comment:
This serializer function name doesn't have a _1 because there's no azure core error response in the input. but the input param type has _1, which should be correct.